File: sc/src/main/java/com/e16din/sc/ILifecycle.java
package com.e16din.sc;

import android.app.Activity;
import android.view.View;
import android.view.ViewGroup;

public interface ILifecycle {

    void beforeBindActivity(Activity activity);

    void onBindActivity(Activity activity);

    void onBindView(View view, ViewGroup vDecor);

    void onShowView(View view);

    void onHideView(View view);

    void onRefresh();
}

File: sc/src/main/java/com/e16din/sc/ILifecycle.kt
package com.e16din.sc

import android.app.Activity
import android.view.View
import android.view.ViewGroup

interface ILifecycle {

    fun beforeBindActivity(activity: Activity)

    fun onBindActivity(activity: Activity)

    fun onBindView(view: View, vDecor: ViewGroup)

    fun onShowView(view: View?)

    fun onHideView(view: View?)
}